typed-racket-lib/typed-racket/typecheck/tc-app/tc-app-lambda.rkt

```diff
@@ -35,7 +35,14 @@
    #:fail-unless (= (syntax-length #'(x ...))
                     (syntax-length #'(args ...))) #f
    #:fail-when (andmap type-annotation (syntax->list #'(x ...))) #f
-   (tc/let-values #'((x) ...) #'(args ...) #'body expected))
+   (let* ([res (tc/let-values #'((x) ...) #'(args ...) #'body expected)]
+          [dom-ts
+           (for/list ([arg (in-list (syntax-e #'(args ...)))])
+             (match (type-of arg)
```

Comment:
Do we really not need annotations from the xs here?

Reply:
Do you mean: should we ask for `(type-of x) ...` instead?
I'm not sure; they might not have a registered type after `tc/let-values` finishes.